### PR TITLE
Update AWS SDK to V3

### PR DESCRIPTION
**Pull request by samfrench**
* Updates the AWS SDK to Version 3 (Currently using V1)
* Updates logic around AWS calls
* Updates specs to test for AWS responses and not mocks of methods in the system being tested
* Make methods not in use by this gem private

### CODE REVIEW

File: lib/alephant/harness/service/s3.rb [hunk 1]
```diff
@@ -1,60 +1,50 @@
...
 
-        def self.exists?(id, object_id)
-          if get_object(id, object_id)
-            yield
-          end
-        end
+        private
 
+        def self.client
+          @client ||= ::Aws::S3::Client.new(AWS.s3_config)
```
Comment (woodyblah) [inline lib/alephant/harness/service/s3.rb]: this used to be a cvar `@@client`, and in the other services it still is. Was this change intentional?

**Review by ettomatic — APPROVED** [review]
